src/EditorFeatures/Core.Wpf/StringIndentation/StringIndentationAdornmentManager.cs

```diff
@@ -39,26 +39,29 @@ protected override void AddAdornmentsToAdornmentLayer_CallOnlyOnUIThread(Normali
 
             foreach (var changedSpan in changedSpanCollection)
             {
+                // Don't do any work for a raw string if it doesn't at least intersect the view span. If it does fully
+                // (or partially) intersect the view, then we'll want to draw the part in view.
                 if (!viewLines.IntersectsBufferSpan(changedSpan))
                     continue;
 
                 var tagSpans = TagAggregator.GetTags(changedSpan);
                 foreach (var tagMappingSpan in tagSpans)
                 {
-                    if (!ShouldDrawTag(changedSpan, tagMappingSpan, out _))
-                        continue;
-
-                    if (!TryMapToSingleSnapshotSpan(tagMappingSpan.Span, TextView.TextSnapshot, out var span))
-                        continue;
-
-                    if (!TryMapHoleSpans(tagMappingSpan.Tag.OrderedHoleSpans, out var orderedHoleSpans))
-                        continue;
-
-                    if (VisibleBlock.CreateVisibleBlock(span, orderedHoleSpans, TextView) is not VisibleBlock block)
+                    // Intentionally not calling TryGetViewLine like other adornment managers do.  That helper ensures
+                    // that the end point of the tag maps to a line that is in view.  However, for raw-string
+                    // indentation adornments it's fine for that point to be offscreen, and we still want to draw the
+                    // indentation line in that case.
+
+                    if (!TryGetMappedPoint(changedSpan, tagMappingSpan, out _) ||
```

Comment:
>if (!TryGetMappedPoint(changedSpan, tagMappingSpan, out _) || [](http://example.com/codeflow?start=20&length=61)

nit: Personal preference is that it's easier to read this many conditions separately, instead of all as one.

Reply:
k

Reply:
done.